src/lib/FreeC/Backend/Agda/Converter/Arg.hs

```diff
@@ -0,0 +1,48 @@
+-- | This module contains functions for generating Agda function, constructor
+--   and type arguments from our intermediate representation.
+
+module FreeC.Backend.Agda.Converter.Arg
+  ( convertTypeVarDecl
+  , lookupIdent
+  , lookupValueIdent
+  , lookupTypeIdent
+  , lookupSmartIdent
```

Comment:
I don't think the `lookup` functions belong into this module. Furthermore, the name `lookupIdent` isn't ideal. It can easily be confused for [`FreeC.Environment.lookupIdent`](https://freeproving.github.io/free-compiler/docs/master/FreeC-Environment.html#v:lookupIdent). Consider to inline the definitions of these functions instead of just moving them to their own module. That would make much more explicit what's actually happening. If that causes expressions to become unreadable, consider to fall back to `do` notation and assign names to intermediate results.